libcontainer/cgroups/utils.go

```diff
@@ -494,6 +494,21 @@ func getHugePageSizeFromFilenames(fileNames []string) ([]string, error) {
 	return pageSizes, nil
 }
 
+// WalkCgroups calls walkFn on the cgroup directory at path and on all its
+// subcgroups.
+func WalkCgroups(path string, walkFn func(path string) error) error {
+	return filepath.Walk(path, func(p string, info os.FileInfo, iErr error) error {
```

Comment:
This is quite expensive as it calls stat(2) on every file. Might be better using filepath.WalkDir() but it's only available in go 1.16+.